### PR TITLE
Remove nix

### PR DESCRIPTION
**Pull request by Warchant**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -16,48 +16,52 @@ env:
...
           persist-credentials: false
-      - uses: cachix/install-nix-action@v24
-        with:
-          github_access_token: ${{ secrets.GITHUB_TOKEN }}
-      - run: nix build --log-lines 100 .\#checks.x86_64-linux.clippy-node-sydney .\#checks.x86_64-linux.nextest-sydney .\#checks.x86_64-linux.clippy-wasm-sydney
-  build-lint-brooklyn:
+
+      - uses: dsherret/rust-toolchain-file@v1
+
+      - name: fmt
```
Comment (akorchyn) [inline .github/workflows/ci.yaml:31]: I would extract fmt to separate job. So the user would have more realistic check-up (including lint)

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -16,48 +16,52 @@ env:
...
+        run: cargo fmt --all -- --check
+      - name: clippy + build
+        run: |
+          packages="runtime-common ggxchain-runtime-brooklyn ggxchain-runtime-sydney"
+          for pkg in $packages; do
+              cargo clippy --package=$pkg -- -D warnings
```
Comment (akorchyn) [inline .github/workflows/ci.yaml:37]: I'm not sure that it will fail if clippy fails but build is successful.
Don't we need to check the exit code?

**Comment by akorchyn** [comment]
I would also include the next lines to the jobs:
```
- uses: technote-space/get-diff-action@v6.1.2
        with:
          PATTERNS: |
            **/**.rs
            Cargo.lock
- name: on each next step
        if: env.GIT_DIFF
```

This way we would not run CI if rust wasn't changed

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (Warchant) [reply .github/workflows/ci.yaml:37]: -D warnings makes it treat warnings as errors

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -16,48 +16,52 @@ env:
...
+              cargo build --locked --package=$pkg --release
```
[the remark this answers is not in the document]
Comment (Warchant) [reply .github/workflows/ci.yaml:38]: they fail :(

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Warchant) [reply .github/workflows/ci.yaml:31]: That would require +1 executor (which may drive costs up)...

**Review by akorchyn — APPROVED** [review]

**Pull request merged**